### PR TITLE
feat(inbox): Add sort by date added to inbox

### PR DESCRIPTION
**Pull request by scttcper**
Sorting by inbox "Date Added" will appear only with the inbox flag and only when the query matches the inbox query. Changing the query to a non-inbox query or clicking a tab will change the sort back to the default (if sorting by inbox time).

The inbox reason badge will display the time it was added to the inbox while sorting by date added.

image

### CODE REVIEW

**Review by wedamija — COMMENTED** [review]
Just a note that we support both `is:unresolved is:for_review` and `is:unresolved is:for_review assigned_or_suggested:me_or_none`

**Comment by scttcper** [comment]
@wedamija nice, both are supported on the frontend

File: src/sentry/static/sentry/app/components/group/inboxBadges/inboxReason.tsx [hunk 1]
```diff
@@ -127,6 +130,7 @@ function InboxReason({inbox, fontSize = 'sm'}: Props) {
               relative: getRelativeDate(dateAdded, 'ago', true),
             }),
         };
+      case GroupInboxReason.NEW:
```
Comment (davidenwang) [inline src/sentry/static/sentry/app/components/group/inboxBadges/inboxReason.tsx:133]: was this just added to be more explicit?

**Review by davidenwang — APPROVED** [review]
lgtm

just want to make sure I understand whats going on: basically the date added sort only gets shown when on the inbox tab and that triggers a new time indicator on the inbox reason? Also moving to another query/tab will remove the date added sort from the list of options right?

File: src/sentry/static/sentry/app/components/group/inboxBadges/inboxReason.tsx (hunk 1 above)
Comment (scttcper) [reply src/sentry/static/sentry/app/components/group/inboxBadges/inboxReason.tsx:133]: i'm making some references to GroupInboxReason.INBOX so i added constants for all of them.

Comment (scttcper) [reply src/sentry/static/sentry/app/components/group/inboxBadges/inboxReason.tsx:133]: oh, yeah. I was looking for NEW and couldn't find it.

**Pull request merged**